src/bundle/Controller/ContentController.php

```diff
@@ -331,4 +333,57 @@ public function previewAction(
             'versionNo' => $versionNo ?? $content->getVersionInfo()->versionNo,
         ]);
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws AdminInvalidArgumentException
+     * @throws \InvalidArgumentException
+     * @throws TranslationInvalidArgumentException
+     * @throws APIRepositoryInvalidArgumentException
+     * @throws UnauthorizedException
+     * @throws InvalidOptionsException
+     */
+    public function updateMainTranslationAction(Request $request): Response
+    {
+        $form = $this->formFactory->updateMainTranslation();
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $result = $this->submitHandler->handle($form, function (MainTranslationUpdateData $data) {
+                $contentInfo = $data->getContentInfo();
+                $mapper = new MainTranslationUpdateMapper();
+                $contentMetadataUpdateStruct = $mapper->reverseMap($data);
+                $this->contentService->updateContentMetadata($contentInfo, $contentMetadataUpdateStruct);
+                $this->notificationHandler->success(
+                    $this->translator->trans(
+                        /** @Desc("Main language for '%name%' updated.") */
+                        'content.main_language_update.success',
+                        ['%name%' => $contentInfo->name],
+                        'content'
+                    )
+                );
+
+                return new RedirectResponse($this->generateUrl('_ezpublishLocation', [
+                    'locationId' => $contentInfo->mainLocationId,
+                    '_fragment' => 'ez-tab-location-view-translations',
+                ]));
+            });
+            if ($result instanceof Response) {
+                return $result;
+            }
+        }
+        /** @var ContentDraftCreateData $data */
```

Comment:
Should be `\EzSystems\EzPlatformAdminUi\Form\Data\Content\Translation\MainTranslationUpdateData`